api/v1beta3/provider_types.go

```diff
@@ -108,6 +108,11 @@ type ProviderSpec struct {
 	// +optional
 	SecretRef *meta.LocalObjectReference `json:"secretRef,omitempty"`
 
+	// ServiceAccountName is the name of the service account used to
+	// authenticate with services from cloud providers.
```

Comment:
Let's add the behavior we discussed here, I like to write good docs for the API struct fields because they show up in the command `kubectl explain provider.spec.serviceAccountName` :pray: